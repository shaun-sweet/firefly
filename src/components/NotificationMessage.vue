<template>
  <q-item class="row notification">
    <div class="device col">{{trigger}}</div>
    <div class="timestamp col row">
      <span class='value'>{{timestamp}}</span>
    </div>
  </q-item>
</template>

<script>
export default {
  props: ['trigger', 'timestamp'],
  data () {
    return {}
  }
}
</script>

<style  lang="stylus" scoped>
</style>
