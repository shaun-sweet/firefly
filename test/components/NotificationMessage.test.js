import Vue from 'vue';
import { shallow } from 'vue-test-utils';
import NotificationMessage from '@/NotificationMessage.vue';

describe('NotificationMessage', () => {
  let wrapper;
  let propsData;
  beforeEach(() => {
    propsData = {
      trigger: "test trigger",
      timestamp: 0
    }
    wrapper = shallow(NotificationMessage,{
      propsData
    });
  })

  it('should render NotificationMessage correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  });
  it('should render the props', () => {
    const triggerDiv = wrapper.find('.device')
    expect(triggerDiv.text()).toBe(propsData.trigger)
    const timestampSpan = wrapper.find('span')
    expect(timestampSpan.text()).toBe(propsData.timestamp.toString())
  })

})
