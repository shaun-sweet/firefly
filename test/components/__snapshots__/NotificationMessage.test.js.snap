// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationMessage should render NotificationMessage correctly 1`] = `
<q-item
  class="row notification"
>
  <div
    class="device col"
  >
    test trigger
  </div>
   
  <div
    class="timestamp col row"
  >
    <span
      class="value"
    >
      0
    </span>
  </div>
</q-item>
`;
